From: Wengling Chen <feiyu2817@gmail.com>
Date: Sat, 2 Nov 2019 08:58:19 +0100
Subject: Remove dependency on
 com.google.android.gms.common/auth/signin/dynamic,
 com.google.android.gms.tasks

---
  chrome/android/BUILD.gn                            |   3 -
  .../chrome/browser/PlayServicesVersionInfo.java    |  11 +-
  chrome/android/javatests/BUILD.gn                  |   3 -
  chrome/android/junit/BUILD.gn                      |   3 -
  chrome/browser/language/android/BUILD.gn           |   1 -
  chrome/browser/omaha/android/BUILD.gn              |   1 -
  .../chrome/browser/omaha/UpdateStatusProvider.java |   4 +-
  chrome/browser/password_manager/android/BUILD.gn   |   5 -
  chrome/browser/ui/android/app_rating/BUILD.gn      |   3 -
  chrome/browser/ui/android/omnibox/BUILD.gn         |   2 -
  chrome/browser/webauthn/android/BUILD.gn           |   1 -
  chrome/browser/webid/BUILD.gn                      |   3 -
  chrome/test/android/BUILD.gn                       |   2 -
  components/externalauth/android/BUILD.gn           |   3 -
  .../components/externalauth/ExternalAuthUtils.java |  12 +-
  .../externalauth/UserRecoverableErrorHandler.java  |  12 +-
  components/media_router/browser/android/BUILD.gn   |   4 -
  .../media_router/BrowserMediaRouter.java           |  21 +--
  components/module_installer/android/BUILD.gn       |   2 -
  .../android/backend/common/BUILD.gn                |   2 -
  .../one_time_tokens/android/backend/sms/BUILD.gn   |   3 -
  components/signin/public/android/BUILD.gn          |   1 -
  components/webauthn/android/BUILD.gn               |   6 -
  content/public/android/BUILD.gn                    |   3 -
  .../content/browser/sms/SmsProviderGms.java        |  10 +-
  content/test/BUILD.gn                              |   3 -
  device/BUILD.gn                                    |   2 -
  services/BUILD.gn                                  |   5 -
  services/device/geolocation/BUILD.gn               |   3 -
  services/shape_detection/BUILD.gn                  |   2 -
  .../BarcodeDetectionProviderImpl.java              |  21 +--
  third_party/android_deps/BUILD.gn                  |   2 -
  third_party/android_deps/autorolled/BUILD.gn       | 164 ---------------------
  .../preconditions/javatests/BUILD.gn               |   1 -
  .../gms/ChromiumPlayServicesAvailability.java      |  12 +-
  third_party/cardboard/BUILD.gn                     |   2 -
  36 files changed, 14 insertions(+), 324 deletions(-)

--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -310,9 +310,6 @@ if (_is_default_toolchain) {
       ":screenshot_provider_java",
       ":usage_stats_proto_java",
       ":xr_java",
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_tasks_java",
       "//base:content_uri_utils_java",
       "//base:flagged_apis_java",
       "//base:process_launcher_java",
--- a/chrome/android/java/src/org/chromium/chrome/browser/PlayServicesVersionInfo.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/PlayServicesVersionInfo.java
@@ -5,12 +5,9 @@
 package org.chromium.chrome.browser;
 
 
-import com.google.android.gms.common.GoogleApiAvailability;
-
 import org.jni_zero.CalledByNative;
 import org.jni_zero.JniType;
 
-import org.chromium.base.PackageUtils;
 import org.chromium.build.annotations.NullMarked;
 import org.chromium.components.externalauth.ExternalAuthUtils;
 
@@ -27,7 +24,7 @@ public class PlayServicesVersionInfo {
      */
     @CalledByNative
     public static @JniType("std::string") String getGmsInfo() {
-        final long sdkVersion = GoogleApiAvailability.GOOGLE_PLAY_SERVICES_VERSION_CODE;
+        final long sdkVersion = 12600000;
         final long installedGmsVersion = getApkVersionNumber();
 
         final String accessType;
@@ -52,11 +49,6 @@ public class PlayServicesVersionInfo {
      *     package is not found.
      */
     public static int getApkVersionNumber() {
-        int ret =
-                PackageUtils.getPackageVersion(GoogleApiAvailability.GOOGLE_PLAY_SERVICES_PACKAGE);
-        if (ret < 0) {
-            ret = 0;
-        }
-        return ret;
+        return 0;
     }
 }
--- a/chrome/android/javatests/BUILD.gn
+++ b/chrome/android/javatests/BUILD.gn
@@ -51,9 +51,6 @@ android_library("chrome_test_java_helper
   ]
 
   public_deps = [
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
-    "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//base:base_java_test_support",
     "//base:base_java_test_support_uncommon",
--- a/chrome/android/junit/BUILD.gn
+++ b/chrome/android/junit/BUILD.gn
@@ -47,9 +47,6 @@ if (is_android) {
     ]
 
     public_deps = [
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_tasks_java",
       "//base:base_java",
       "//base:base_java_test_support",
       "//base:base_java_test_support_uncommon",
--- a/chrome/browser/language/android/BUILD.gn
+++ b/chrome/browser/language/android/BUILD.gn
@@ -13,7 +13,6 @@ android_library("base_module_java") {
     "java/src/org/chromium/chrome/browser/language/LanguageSplitInstaller.java",
   ]
   deps = [
-    "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//build/android:build_java",
     "//chrome/browser/preferences:java",
--- a/chrome/browser/omaha/android/BUILD.gn
+++ b/chrome/browser/omaha/android/BUILD.gn
@@ -21,7 +21,6 @@ android_library("java") {
   ]
   deps = [
     ":update_proto_java",
-    "$google_play_services_package:google_play_services_base_java",
     "//base:base_java",
     "//base:base_shared_preferences_java",
     "//base/version_info/android:version_constants_java",
--- a/chrome/browser/omaha/android/java/src/org/chromium/chrome/browser/omaha/UpdateStatusProvider.java
+++ b/chrome/browser/omaha/android/java/src/org/chromium/chrome/browser/omaha/UpdateStatusProvider.java
@@ -16,8 +16,6 @@ import android.text.TextUtils;
 
 import androidx.annotation.IntDef;
 
-import com.google.android.gms.common.GooglePlayServicesUtil;
-
 import org.chromium.base.ApkInfo;
 import org.chromium.base.Callback;
 import org.chromium.base.DeviceInfo;
@@ -301,7 +299,7 @@ public class UpdateStatusProvider {
                                 // Disable the version update check for automotive. See b/297925838.
                                 && !DeviceInfo.isAutomotive()
                                 && PackageUtils.isPackageInstalled(
-                                        GooglePlayServicesUtil.GOOGLE_PLAY_STORE_PACKAGE);
+                                        "com.android.vending");
                 status.updateState =
                         allowedToUpdate ? UpdateState.UPDATE_AVAILABLE : UpdateState.NONE;
 
--- a/chrome/browser/password_manager/android/BUILD.gn
+++ b/chrome/browser/password_manager/android/BUILD.gn
@@ -118,8 +118,6 @@ android_library("java") {
     ":java_resources",
     ":settings_interface_java",
     ":utils_java",
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java",
     "//build/android:build_java",
     "//chrome/browser/device_reauth/android:java",
@@ -225,7 +223,6 @@ android_library("utils_java") {
 
   deps = [
     ":java_resources",
-    "$google_play_services_package:google_play_services_base_java",
     "//base:base_java",
     "//base:resetters_java",
     "//base:tasks_java",
@@ -301,8 +298,6 @@ robolectric_binary("password_manager_jun
     ":test_support_java",
     ":unified_password_manager_proto_java",
     ":utils_java",
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
     "add_username_dialog:junit",
     "bottom_sheet:junit",
     "grouped_affiliations:junit",
--- a/chrome/browser/ui/android/app_rating/BUILD.gn
+++ b/chrome/browser/ui/android/app_rating/BUILD.gn
@@ -13,8 +13,6 @@ if (enable_chrome_android_internal) {
     # Skip core_common_java to avoid resource collisions with clank's internal
     # playcore.
     deps = [
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_tasks_java",
     ]
   }
 
@@ -53,7 +51,6 @@ android_library("impl_java") {
   proguard_configs = [ "proguard.flags" ]
   deps = [
     ":java",
-    "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//build/android:build_java",
     "//third_party/androidx:androidx_annotation_annotation_java",
--- a/chrome/browser/ui/android/omnibox/BUILD.gn
+++ b/chrome/browser/ui/android/omnibox/BUILD.gn
@@ -212,7 +212,6 @@ android_library("java") {
 
   deps = [
     ":java_resources",
-    "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//base:supplier_java",
     "//base/version_info/android:version_constants_java",
@@ -612,7 +611,6 @@ robolectric_library("junit") {
   deps = [
     ":java",
     ":java_resources",
-    "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//base:base_java_test_support",
     "//base:base_junit_test_support",
--- a/chrome/browser/webauthn/android/BUILD.gn
+++ b/chrome/browser/webauthn/android/BUILD.gn
@@ -16,7 +16,6 @@ android_library("java") {
 
   deps = [
     ":java_resources",
-    "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//build/android:build_java",
     "//chrome/android:chrome_app_java_resources",
--- a/chrome/browser/webid/BUILD.gn
+++ b/chrome/browser/webid/BUILD.gn
@@ -19,8 +19,6 @@ if (is_android) {
       "android/java/src/org/chromium/chrome/browser/webid/VerifiedOriginResolver.java",
     ]
     deps = [
-      "$google_play_services_package:google_play_services_identity_credentials_java",
-      "$google_play_services_package:google_play_services_tasks_java",
       "//base:base_java",
       "//base:service_loader_java",
       "//chrome/browser/android/browserservices/verification:java",
@@ -57,7 +55,6 @@ if (is_android) {
 
     deps = [
       ":java",
-      "$google_play_services_package:google_play_services_identity_credentials_java",
       "//base:base_junit_test_support",
       "//base:promise_java",
       "//content/public/android:content_java",
--- a/chrome/test/android/BUILD.gn
+++ b/chrome/test/android/BUILD.gn
@@ -399,8 +399,6 @@ android_library("chrome_java_integration
 
   deps = [
     ":chrome_java_test_support_common",
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java",
     "//base:base_java_test_support",
     "//base:holder_java",
--- a/components/externalauth/android/BUILD.gn
+++ b/components/externalauth/android/BUILD.gn
@@ -6,8 +6,6 @@ import("//build/config/android/rules.gni
 
 android_library("java") {
   deps = [
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java",
     "//base:service_loader_java",
     "//components/embedder_support/android:util_java",
@@ -25,7 +23,6 @@ android_library("java") {
 robolectric_library("junit") {
   deps = [
     ":java",
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java",
     "//base:base_java_test_support",
     "//base:base_junit_test_support",
--- a/components/externalauth/android/java/src/org/chromium/components/externalauth/ExternalAuthUtils.java
+++ b/components/externalauth/android/java/src/org/chromium/components/externalauth/ExternalAuthUtils.java
@@ -17,9 +17,6 @@ import android.text.TextUtils;
 import androidx.annotation.VisibleForTesting;
 import androidx.annotation.WorkerThread;
 
-import com.google.android.gms.common.ConnectionResult;
-import com.google.android.gms.common.GoogleApiAvailability;
-
 import org.chromium.base.ContextUtils;
 import org.chromium.base.Log;
 import org.chromium.base.ResettersForTesting;
@@ -176,9 +173,7 @@ public class ExternalAuthUtils {
      *         when it is updating.
      */
     public boolean isGooglePlayServicesMissing(final Context context) {
-        final int resultCode = checkGooglePlayServicesAvailable(context);
-        return (resultCode == ConnectionResult.SERVICE_MISSING
-                || resultCode == ConnectionResult.SERVICE_INVALID);
+        return true;
     }
 
     /**
@@ -195,7 +190,6 @@ public class ExternalAuthUtils {
     public boolean canUseGooglePlayServices(final UserRecoverableErrorHandler errorHandler) {
         Context context = ContextUtils.getApplicationContext();
         final int resultCode = checkGooglePlayServicesAvailable(context);
-        if (resultCode == ConnectionResult.SUCCESS) return true;
         // resultCode is some kind of error.
         Log.v(TAG, "Unable to use Google Play Services: %s", describeError(resultCode));
         if (isUserRecoverableError(resultCode)) {
@@ -276,7 +270,7 @@ public class ExternalAuthUtils {
      * @return true If the code represents a user-recoverable error
      */
     protected boolean isUserRecoverableError(final int errorCode) {
-        return GoogleApiAvailability.getInstance().isUserResolvableError(errorCode);
+        return false;
     }
 
     /**
@@ -286,7 +280,7 @@ public class ExternalAuthUtils {
      * @return a textual description of the error code
      */
     protected String describeError(final int errorCode) {
-        return GoogleApiAvailability.getInstance().getErrorString(errorCode);
+        return "";
     }
 
     /**
--- a/components/externalauth/android/java/src/org/chromium/components/externalauth/UserRecoverableErrorHandler.java
+++ b/components/externalauth/android/java/src/org/chromium/components/externalauth/UserRecoverableErrorHandler.java
@@ -9,8 +9,6 @@ import android.app.Dialog;
 import android.content.Context;
 import android.content.DialogInterface;
 
-import com.google.android.gms.common.GoogleApiAvailability;
-
 import org.chromium.base.ThreadUtils;
 import org.chromium.base.metrics.RecordUserAction;
 import org.chromium.build.annotations.NullMarked;
@@ -90,7 +88,6 @@ public abstract class UserRecoverableErr
             if (!sNotificationShown.getAndSet(true)) {
                 return;
             }
-            GoogleApiAvailability.getInstance().showErrorNotification(context, errorCode);
         }
     }
 
@@ -178,17 +175,10 @@ public abstract class UserRecoverableErr
             }
             Dialog dialog = mDialog;
             if (dialog == null) {
-                dialog =
-                        GoogleApiAvailability.getInstance()
-                                .getErrorDialog(mActivity, errorCode, NO_RESPONSE_REQUIRED);
-                assert dialog != null : "code was " + errorCode;
-                mDialog = dialog;
                 mErrorCode = errorCode;
-
-                DialogUserActionRecorder.createAndAttachToDialog(dialog);
             }
             // This can happen if |errorCode| is ConnectionResult.SERVICE_INVALID.
-            if (!dialog.isShowing()) {
+            if (dialog != null && !dialog.isShowing()) {
                 dialog.setCancelable(mCancelable);
                 dialog.show();
                 RecordUserAction.record("Signin_Android_GmsUserRecoverableDialogShown");
--- a/components/media_router/browser/android/BUILD.gn
+++ b/components/media_router/browser/android/BUILD.gn
@@ -12,8 +12,6 @@ android_library("java") {
   resources_package = "org.chromium.components.media_router"
   deps = [
     ":java_resources",
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java",
     "//build/android:build_java",
     "//components/browser_ui/media/android:java",
@@ -143,8 +141,6 @@ robolectric_library("junit") {
   deps = [
     ":java",
     ":test_support_java",
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java",
     "//base:base_java_test_support",
     "//base:base_junit_test_support",
--- a/components/media_router/browser/android/java/src/org/chromium/components/media_router/BrowserMediaRouter.java
+++ b/components/media_router/browser/android/java/src/org/chromium/components/media_router/BrowserMediaRouter.java
@@ -9,9 +9,6 @@ import static org.chromium.build.NullUti
 import androidx.annotation.VisibleForTesting;
 import androidx.mediarouter.media.MediaRouter;
 
-import com.google.android.gms.common.ConnectionResult;
-import com.google.android.gms.common.GoogleApiAvailability;
-
 import org.jni_zero.CalledByNative;
 import org.jni_zero.JNINamespace;
 import org.jni_zero.JniType;
@@ -47,23 +44,7 @@ public class BrowserMediaRouter implemen
             new MediaRouteProvider.Factory() {
                 @Override
                 public void addProviders(MediaRouteManager manager) {
-                    int googleApiAvailabilityResult =
-                            GoogleApiAvailability.getInstance()
-                                    .isGooglePlayServicesAvailable(
-                                            ContextUtils.getApplicationContext(),
-                                            MIN_GOOGLE_PLAY_SERVICES_APK_VERSION);
-                    if (googleApiAvailabilityResult != ConnectionResult.SUCCESS) {
-                        GoogleApiAvailability.getInstance()
-                                .showErrorNotification(
-                                        ContextUtils.getApplicationContext(),
-                                        googleApiAvailabilityResult);
-                        return;
-                    }
-                    MediaRouteProvider cafProvider = CafMediaRouteProvider.create(manager);
-                    manager.addMediaRouteProvider(cafProvider);
-                    MediaRouteProvider remotingProvider =
-                            CafRemotingMediaRouteProvider.create(manager);
-                    manager.addMediaRouteProvider(remotingProvider);
+                    return;
                 }
             };
 
--- a/components/module_installer/android/BUILD.gn
+++ b/components/module_installer/android/BUILD.gn
@@ -26,7 +26,6 @@ android_library("module_installer_java")
   ]
 
   deps = [
-    "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//components/crash/android:java",
     "//third_party/android_deps:playcore_java",
@@ -47,7 +46,6 @@ robolectric_binary("module_installer_jun
   ]
   deps = [
     ":module_installer_java",
-    "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//base:base_java_test_support",
     "//base:base_junit_test_support",
--- a/components/one_time_tokens/android/backend/common/BUILD.gn
+++ b/components/one_time_tokens/android/backend/common/BUILD.gn
@@ -11,7 +11,6 @@ android_library("java") {
     "java/src/org/chromium/components/one_time_tokens/backend/OneTimeTokensMetricsRecorder.java",
   ]
   deps = [
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java",
     "//build/android:build_java",
     "//third_party/androidx:androidx_annotation_annotation_java",
@@ -23,7 +22,6 @@ robolectric_library("junit") {
   sources = [ "java/src/org/chromium/components/one_time_tokens/backend/OneTimeTokensMetricsRecorderTest.java" ]
   deps = [
     ":java",
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java",
     "//base:base_java_test_support",
     "//base:base_junit_test_support",
--- a/components/one_time_tokens/android/backend/sms/BUILD.gn
+++ b/components/one_time_tokens/android/backend/sms/BUILD.gn
@@ -48,7 +48,6 @@ android_library("java") {
   ]
 
   deps = [
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java",
     "//build/android:build_java",
     "//components/one_time_tokens/android/backend/common:java",
@@ -73,7 +72,6 @@ robolectric_binary("one_time_tokens_juni
   deps = [
     ":java",
     ":sms_otp_fetching_interface_java",
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java",
     "//base:base_java_test_support",
     "//base:base_junit_test_support",
@@ -85,7 +83,6 @@ robolectric_binary("one_time_tokens_juni
 
 android_library("sms_otp_fetching_interface_java") {
   deps = [
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java",
     "//base:service_loader_java",
   ]
--- a/components/signin/public/android/BUILD.gn
+++ b/components/signin/public/android/BUILD.gn
@@ -3,7 +3,6 @@ import("//third_party/jni_zero/jni_zero.
 
 android_library("java") {
   deps = [
-    "$google_play_services_package:google_play_services_base_java",
     "//base:base_java",
     "//base:service_loader_java",
     "//build/android:build_java",
--- a/components/webauthn/android/BUILD.gn
+++ b/components/webauthn/android/BUILD.gn
@@ -75,10 +75,6 @@ android_library("java") {
   ]
 
   deps = [
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
-    "$google_play_services_package:google_play_services_identity_credentials_java",
-    "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//base:service_loader_java",
     "//base/version_info/android:version_constants_java",
@@ -232,8 +228,6 @@ robolectric_library("junit") {
   deps = [
     ":java",
     ":test_support_java",
-    "$google_play_services_package:google_play_services_identity_credentials_java",
-    "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//base:base_java_test_support",
     "//base:base_junit_test_support",
--- a/content/public/android/BUILD.gn
+++ b/content/public/android/BUILD.gn
@@ -163,9 +163,6 @@ android_library("content_full_java") {
     ":content_java_resources",
     ":content_main_dex_java",
     ":content_util_java",
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
-    "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//base:flagged_apis_java",
     "//base:process_launcher_java",
--- a/content/public/android/java/src/org/chromium/content/browser/sms/SmsProviderGms.java
+++ b/content/public/android/java/src/org/chromium/content/browser/sms/SmsProviderGms.java
@@ -6,9 +6,6 @@ package org.chromium.content.browser.sms
 
 import androidx.annotation.VisibleForTesting;
 
-import com.google.android.gms.common.ConnectionResult;
-import com.google.android.gms.common.GoogleApiAvailability;
-
 import org.jni_zero.CalledByNative;
 import org.jni_zero.JNINamespace;
 import org.jni_zero.NativeMethods;
@@ -92,12 +89,7 @@ public class SmsProviderGms {
     @CalledByNative
     private static SmsProviderGms create(long smsProviderGmsAndroid, @GmsBackend int backend) {
         Log.d(TAG, "Creating SmsProviderGms");
-        boolean isVerificationBackendAvailable =
-                GoogleApiAvailability.getInstance()
-                                .isGooglePlayServicesAvailable(
-                                        ContextUtils.getApplicationContext(),
-                                        MIN_GMS_VERSION_NUMBER_WITH_CODE_BROWSER_BACKEND)
-                        == ConnectionResult.SUCCESS;
+        boolean isVerificationBackendAvailable = false;
         return new SmsProviderGms(smsProviderGmsAndroid, backend, isVerificationBackendAvailable);
     }
 
--- a/content/test/BUILD.gn
+++ b/content/test/BUILD.gn
@@ -3811,9 +3811,6 @@ if (is_android) {
     testonly = true
     sources = content_java_sources_needing_jni
     deps = [
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_tasks_java",
       "//base:base_java",
       "//content/public/android:content_java",
       "//third_party/jni_zero:jni_zero_java",
--- a/device/BUILD.gn
+++ b/device/BUILD.gn
@@ -538,8 +538,6 @@ if (is_android) {
       "vr/public/junit/src/org/chromium/device/vr/XrFeatureStatusTest.java",
     ]
     deps = [
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
       "//base:base_java",
       "//base:base_java_test_support",
       "//base:base_junit_test_support",
--- a/services/BUILD.gn
+++ b/services/BUILD.gn
@@ -204,9 +204,6 @@ if (is_android) {
       "shape_detection/android/junit/src/org/chromium/shape_detection/BitmapUtilsTest.java",
     ]
     deps = [
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_tasks_java",
       "//base:base_java",
       "//base:base_java_test_support",
       "//base:base_junit_test_support",
@@ -246,8 +243,6 @@ if (is_android) {
       "shape_detection/android/javatests/src/org/chromium/shape_detection/TextDetectionImplTest.java",
     ]
     deps = [
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
       "//base:base_java",
       "//base:base_java_test_support",
       "//mojo/public/java:base_java",
--- a/services/device/geolocation/BUILD.gn
+++ b/services/device/geolocation/BUILD.gn
@@ -160,9 +160,6 @@ if (is_android) {
 
     deps = [
       ":geolocation_jni_headers",
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_tasks_java",
       "//base:base_java",
       "//build/android:build_java",
       "//components/location/android:location_java",
--- a/services/shape_detection/BUILD.gn
+++ b/services/shape_detection/BUILD.gn
@@ -130,8 +130,6 @@ if (is_android) {
     ]
 
     deps = [
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
       "//base:base_java",
       "//mojo/public/java:base_java",
       "//mojo/public/java:bindings_java",
--- a/services/shape_detection/android/java/src/org/chromium/shape_detection/BarcodeDetectionProviderImpl.java
+++ b/services/shape_detection/android/java/src/org/chromium/shape_detection/BarcodeDetectionProviderImpl.java
@@ -7,8 +7,6 @@ import org.chromium.build.annotations.Nu
 
 import android.content.Context;
 
-import com.google.android.gms.common.GoogleApiAvailability;
-
 import org.chromium.base.ContextUtils;
 import org.chromium.base.Log;
 import org.chromium.base.PackageUtils;
@@ -64,22 +62,7 @@ public class BarcodeDetectionProviderImp
     public void onConnectionError(MojoException e) {}
 
     public static @Nullable BarcodeDetectionProvider create() {
-        Context ctx = ContextUtils.getApplicationContext();
-        if (!ChromiumPlayServicesAvailability.isGooglePlayServicesAvailable(ctx)) {
-            Log.w(TAG, "Google Play Services not available");
-            return null;
-        }
-        int version =
-                PackageUtils.getPackageVersion(GoogleApiAvailability.GOOGLE_PLAY_SERVICES_PACKAGE);
-        if (version < 19742000) {
-            if (version < 0) {
-                Log.w(TAG, "Google Play Services not available");
-            } else {
-                // https://crbug.com/1020746
-                Log.w(TAG, "Detection disabled (%d < 19.7.42)", version);
-            }
-            return null;
-        }
-        return new BarcodeDetectionProviderImpl();
+        Log.w(TAG, "Google Play Services not available");
+        return null;
     }
 }
--- a/third_party/android_deps/BUILD.gn
+++ b/third_party/android_deps/BUILD.gn
@@ -92,8 +92,6 @@ if (!limit_android_deps) {
       sources =
           [ "util/org/chromium/gms/ChromiumPlayServicesAvailability.java" ]
       deps = [
-        "$google_play_services_package:google_play_services_base_java",
-        "$google_play_services_package:google_play_services_basement_java",
         "//base:resetters_java",
       ]
     }
--- a/third_party/android_deps/autorolled/BUILD.gn
+++ b/third_party/android_deps/autorolled/BUILD.gn
@@ -74,93 +74,6 @@ template("define_autorolled_targets") {
 
     # This is generated, do not edit. Update BuildConfigGenerator.groovy
     # instead.
-    if (google_play_services_package == "//third_party/android_deps") {
-      android_aar_prebuilt("google_play_services_base_java") {
-        aar_path = "autorolled/cipd/libs/com_google_android_gms_play_services_base/play-services-base.aar"
-        info_path = "autorolled/committed/libs/com_google_android_gms_play_services_base/com_google_android_gms_play_services_base.info"
-        enable_bytecode_checks = false
-        deps = [
-          "$google_play_services_package:google_play_services_basement_java",
-          "$google_play_services_package:google_play_services_tasks_java",
-          "//third_party/androidx:androidx_collection_collection_java",
-          "//third_party/androidx:androidx_core_core_java",
-          "//third_party/androidx:androidx_fragment_fragment_java",
-        ]
-      }
-    }
-
-    # This is generated, do not edit. Update BuildConfigGenerator.groovy
-    # instead.
-    if (google_play_services_package == "//third_party/android_deps") {
-      android_aar_prebuilt("google_play_services_basement_java") {
-        aar_path = "autorolled/cipd/libs/com_google_android_gms_play_services_basement/play-services-basement.aar"
-        info_path = "autorolled/committed/libs/com_google_android_gms_play_services_basement/com_google_android_gms_play_services_basement.info"
-        enable_bytecode_checks = false
-        deps = [
-          "//third_party/androidx:androidx_collection_collection_java",
-          "//third_party/androidx:androidx_core_core_java",
-          "//third_party/androidx:androidx_fragment_fragment_java",
-        ]
-
-        jar_excluded_patterns = []
-        if (!enable_java_asserts) {
-          # Omit the file since we use our own copy.
-          jar_excluded_patterns +=
-              [ "com/google/android/gms/common/internal/Preconditions.class" ]
-          deps += [ "//third_party/android_deps/local_modifications/preconditions:gms_stub_preconditions_java" ]
-        }
-
-        # https://crbug.com/989505
-        jar_excluded_patterns += [ "META-INF/proguard/*" ]
-      }
-    }
-
-    # This is generated, do not edit. Update BuildConfigGenerator.groovy
-    # instead.
-    if (google_play_services_package == "//third_party/android_deps") {
-      android_aar_prebuilt("google_play_services_identity_credentials_java") {
-        aar_path = "autorolled/cipd/libs/com_google_android_gms_play_services_identity_credentials/play-services-identity-credentials.aar"
-        info_path = "autorolled/committed/libs/com_google_android_gms_play_services_identity_credentials/com_google_android_gms_play_services_identity_credentials.info"
-        enable_bytecode_checks = false
-        deps = [
-          "$google_play_services_package:google_play_services_base_java",
-          "$google_play_services_package:google_play_services_basement_java",
-          "$google_play_services_package:google_play_services_tasks_java",
-          "//third_party/kotlin_stdlib:kotlin_stdlib_java",
-        ]
-      }
-    }
-
-    # This is generated, do not edit. Update BuildConfigGenerator.groovy
-    # instead.
-    if (google_play_services_package == "//third_party/android_deps") {
-      android_aar_prebuilt("google_play_services_tasks_java") {
-        aar_path = "autorolled/cipd/libs/com_google_android_gms_play_services_tasks/play-services-tasks.aar"
-        info_path = "autorolled/committed/libs/com_google_android_gms_play_services_tasks/com_google_android_gms_play_services_tasks.info"
-        enable_bytecode_checks = false
-        deps = [
-          "$google_play_services_package:google_play_services_basement_java",
-        ]
-      }
-    }
-
-    # This is generated, do not edit. Update BuildConfigGenerator.groovy
-    # instead.
-    if (google_play_services_package == "//third_party/android_deps") {
-      android_aar_prebuilt("google_play_services_time_java") {
-        aar_path = "autorolled/cipd/libs/com_google_android_gms_play_services_time/play-services-time.aar"
-        info_path = "autorolled/committed/libs/com_google_android_gms_play_services_time/com_google_android_gms_play_services_time.info"
-        enable_bytecode_checks = false
-        deps = [
-          "$google_play_services_package:google_play_services_base_java",
-          "$google_play_services_package:google_play_services_basement_java",
-          "$google_play_services_package:google_play_services_tasks_java",
-        ]
-      }
-    }
-
-    # This is generated, do not edit. Update BuildConfigGenerator.groovy
-    # instead.
     if (!defined(material_design_target)) {
       android_aar_prebuilt("com_google_android_material_material_java") {
         aar_path = "autorolled/cipd/libs/com_google_android_material_material/material.aar"
@@ -353,9 +266,6 @@ template("define_autorolled_targets") {
       info_path = "autorolled/committed/libs/com_google_mlkit_common/com_google_mlkit_common.info"
       enable_bytecode_checks = false
       deps = [
-        "$google_play_services_package:google_play_services_base_java",
-        "$google_play_services_package:google_play_services_basement_java",
-        "$google_play_services_package:google_play_services_tasks_java",
         "//third_party/androidx:androidx_appcompat_appcompat_java",
         "//third_party/androidx:androidx_core_core_java",
       ]
@@ -369,7 +279,6 @@ template("define_autorolled_targets") {
       info_path = "autorolled/committed/libs/com_google_mlkit_genai_common/com_google_mlkit_genai_common.info"
       enable_bytecode_checks = false
       deps = [
-        "$google_play_services_package:google_play_services_basement_java",
         "//third_party/android_deps:org_jetbrains_kotlinx_kotlinx_coroutines_core_java",
         "//third_party/android_deps:org_jetbrains_kotlinx_kotlinx_coroutines_guava_java",
         "//third_party/kotlin_stdlib:kotlin_stdlib_java",
@@ -384,8 +293,6 @@ template("define_autorolled_targets") {
       info_path = "autorolled/committed/libs/com_google_mlkit_genai_prompt/com_google_mlkit_genai_prompt.info"
       enable_bytecode_checks = false
       deps = [
-        "$google_play_services_package:google_play_services_basement_java",
-        "$google_play_services_package:google_play_services_tasks_java",
         "//third_party/android_deps:com_google_mlkit_common_java",
         "//third_party/android_deps:com_google_mlkit_genai_common_java",
         "//third_party/android_deps:com_google_mlkit_genai_schema_java",
@@ -761,51 +668,6 @@ template("define_autorolled_targets") {
 
     # This is generated, do not edit. Update BuildConfigGenerator.groovy
     # instead.
-    if (google_play_services_package == "//third_party/android_deps") {
-      android_aar_prebuilt("google_play_services_cloud_messaging_java") {
-        aar_path = "autorolled/cipd/libs/com_google_android_gms_play_services_cloud_messaging/play-services-cloud-messaging.aar"
-        info_path = "autorolled/committed/libs/com_google_android_gms_play_services_cloud_messaging/com_google_android_gms_play_services_cloud_messaging.info"
-        enable_bytecode_checks = false
-
-        # To remove visibility constraint, add this dependency to
-        # //third_party/android_deps/autorolled/build.gradle.template
-        visibility = [
-          ":*",
-          "//third_party/androidx:*",
-        ]
-        deps = [
-          "$google_play_services_package:google_play_services_base_java",
-          "$google_play_services_package:google_play_services_basement_java",
-          "$google_play_services_package:google_play_services_tasks_java",
-        ]
-      }
-    }
-
-    # This is generated, do not edit. Update BuildConfigGenerator.groovy
-    # instead.
-    if (google_play_services_package == "//third_party/android_deps") {
-      android_aar_prebuilt("google_play_services_phenotype_java") {
-        aar_path = "autorolled/cipd/libs/com_google_android_gms_play_services_phenotype/play-services-phenotype.aar"
-        info_path = "autorolled/committed/libs/com_google_android_gms_play_services_phenotype/com_google_android_gms_play_services_phenotype.info"
-        enable_bytecode_checks = false
-
-        # To remove visibility constraint, add this dependency to
-        # //third_party/android_deps/autorolled/build.gradle.template
-        visibility = [
-          ":*",
-          "//third_party/androidx:*",
-        ]
-        deps = [
-          "$google_play_services_package:google_play_services_base_java",
-          "$google_play_services_package:google_play_services_basement_java",
-          "$google_play_services_package:google_play_services_tasks_java",
-          "//third_party/androidx:androidx_core_core_java",
-        ]
-      }
-    }
-
-    # This is generated, do not edit. Update BuildConfigGenerator.groovy
-    # instead.
     android_aar_prebuilt("com_google_android_play_core_common_java") {
       aar_path = "autorolled/cipd/libs/com_google_android_play_core_common/core-common.aar"
       info_path = "autorolled/committed/libs/com_google_android_play_core_common/com_google_android_play_core_common.info"
@@ -1493,29 +1355,6 @@ template("define_autorolled_targets") {
     }
 
     # This is generated, do not edit. Update BuildConfigGenerator.groovy
-    # instead.
-    java_prebuilt(
-        "org_jetbrains_kotlinx_kotlinx_coroutines_play_services_java") {
-      jar_path = "autorolled/cipd/libs/org_jetbrains_kotlinx_kotlinx_coroutines_play_services/kotlinx-coroutines-play-services.jar"
-      output_name = "org_jetbrains_kotlinx_kotlinx_coroutines_play_services"
-      supports_android = true
-      requires_android = true
-      enable_bytecode_checks = false
-
-      # To remove visibility constraint, add this dependency to
-      # //third_party/android_deps/autorolled/build.gradle.template
-      visibility = [
-        ":*",
-        "//third_party/androidx:*",
-      ]
-      deps = [
-        "$google_play_services_package:google_play_services_tasks_java",
-        "//third_party/android_deps:org_jetbrains_kotlinx_kotlinx_coroutines_core_java",
-        "//third_party/kotlin_stdlib:kotlin_stdlib_java",
-      ]
-    }
-
-    # This is generated, do not edit. Update BuildConfigGenerator.groovy
     # instead.
     java_group("org_jetbrains_kotlinx_kotlinx_coroutines_test_java") {
       # To remove visibility constraint, add this dependency to
--- a/third_party/android_deps/local_modifications/preconditions/javatests/BUILD.gn
+++ b/third_party/android_deps/local_modifications/preconditions/javatests/BUILD.gn
@@ -8,7 +8,6 @@ android_library("javatests") {
   testonly = true
   sources = [ "org/chromium/preconditions/PreconditionsTest.java" ]
   deps = [
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java_test_support",
     "//base/test:test_support_java",
     "//build/android:build_java",
--- a/third_party/android_deps/util/org/chromium/gms/ChromiumPlayServicesAvailability.java
+++ b/third_party/android_deps/util/org/chromium/gms/ChromiumPlayServicesAvailability.java
@@ -5,9 +5,6 @@ package org.chromium.gms;
 
 import android.content.Context;
 
-import com.google.android.gms.common.ConnectionResult;
-import com.google.android.gms.common.GoogleApiAvailability;
-
 import org.chromium.build.annotations.NullMarked;
 import org.chromium.build.annotations.Nullable;
 import org.chromium.base.ResettersForTesting;
@@ -45,9 +42,7 @@ public final class ChromiumPlayServicesA
         if (sIsAvailableForTesting != null) {
             return sIsAvailableForTesting;
         }
-        return GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(
-                       context, GMS_VERSION_NUMBER)
-                == ConnectionResult.SUCCESS;
+        return false;
     }
 
     /**
@@ -66,9 +61,8 @@ public final class ChromiumPlayServicesA
      */
     public static int getGooglePlayServicesConnectionResult(final Context context) {
         if (sIsAvailableForTesting != null) {
-            return sIsAvailableForTesting ? ConnectionResult.SUCCESS : ConnectionResult.SERVICE_VERSION_UPDATE_REQUIRED;
+            return sIsAvailableForTesting ? 0 /* SUCCESS */ : 2 /* SERVICE_VERSION_UPDATE_REQUIRED */;
         }
-        return GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(
-                context, GMS_VERSION_NUMBER);
+        return 1; // SERVICE_MISSING
     }
 }
--- a/third_party/cardboard/BUILD.gn
+++ b/third_party/cardboard/BUILD.gn
@@ -51,8 +51,6 @@ android_library("cardboard_java") {
   deps = [
     ":cardboard_proto_java",
     ":cardboard_resources",
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
     "//third_party/android_deps:protobuf_lite_runtime_java",
     "//third_party/androidx:androidx_annotation_annotation_jvm_java",
     "//third_party/androidx:androidx_appcompat_appcompat_java",
